pytest_tests/tests/services/rest_gate/test_rest_gate_new.py

```diff
@@ -0,0 +1,524 @@
+                return_response=True,
+                download=True,
+            )
+
+            assert "attachment" in resp.headers["Content-Disposition"]
+
+    @allure.title("Verify Content-Type if uploaded without any Content-Type specified")
+    def test_put_http_get_http_without_content_type(self, simple_object_size, gw_params):
+        cid = create_container(
+            self.wallet.path,
+            shell=self.shell,
+            endpoint=self.neofs_env.sn_rpc,
+            rule=self.PLACEMENT_RULE_2,
+            basic_acl=PUBLIC_ACL,
+        )
+
+        with allure.step("Upload binary object"):
+            file_path = generate_file(simple_object_size)
+
+            oid = new_upload_via_rest_gate(
+                cid=cid,
+                path=file_path,
+                endpoint=gw_params["endpoint"],
+            )
+
+            resp = get_via_rest_gate(
+                cid=cid,
+                oid=oid,
+                endpoint=gw_params["endpoint"],
+                return_response=True,
+            )
+            assert resp.headers["Content-Type"] == "application/octet-stream"
+
+        with allure.step("Upload text object"):
+            file_path = generate_file_with_content(simple_object_size, content="123")
+
+            oid = new_upload_via_rest_gate(
+                cid=cid,
+                path=file_path,
+                endpoint=gw_params["endpoint"],
+            )
+
+            resp = get_via_rest_gate(
+                cid=cid,
+                oid=oid,
+                endpoint=gw_params["endpoint"],
+                return_response=True,
+            )
+            assert resp.headers["Content-Type"] == "text/plain; charset=utf-8"
+
+    @allure.title("Verify Content-Type if uploaded with X-Attribute-Content-Type")
+    def test_put_http_get_http_with_x_atribute_content_type(self, simple_object_size, gw_params):
+        cid = create_container(
+            self.wallet.path,
+            shell=self.shell,
+            endpoint=self.neofs_env.sn_rpc,
+            rule=self.PLACEMENT_RULE_2,
+            basic_acl=PUBLIC_ACL,
+        )
+
+        with allure.step("Upload object with X-Attribute-Content-Type"):
+            file_path = generate_file(simple_object_size)
+
+            headers = {"X-Attributes": '{"Content-Type":"CoolContentType"}'}
+            oid = new_upload_via_rest_gate(
+                cid=cid,
+                path=file_path,
+                headers=headers,
+                endpoint=gw_params["endpoint"],
+            )
+
+            resp = get_via_rest_gate(
+                cid=cid,
+                oid=oid,
+                endpoint=gw_params["endpoint"],
+                return_response=True,
+            )
+            assert resp.headers["Content-Type"] == "CoolContentType"
+
+    @allure.title("Verify Content-Type if uploaded with multipart Content-Type")
+    def test_put_http_get_http_with_multipart_content_type(self, gw_params):
```

Comment:
What is so `multipart` here?

Reply:
Nothing really, it's a copy-paste from the same function in the old upload, and there wasn't any multipart either. I'd better rename both of them.

Reply:
Done.